src/coord/src/catalog.rs

```diff
@@ -462,6 +474,12 @@ impl CatalogEntry {
         &self.item
     }
 
+    /// Returns a mutable reference to the `CatalogItem` associated with this
+    /// catalog entry.
+    pub fn item_mut(&mut self) -> &mut CatalogItem {
```

Comment:
this is becuase StreamWriteHandle::write is &mut self. it actually doesn't need to be but it's like that because semantically it probably should be. how much do you care about all the &muts this adds to catalog stuff